flutter/lib/src/widgets_binding_observer.dart

```diff
@@ -0,0 +1,155 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+/// This is a `WidgetsBindingObserver` which can observe some events of a
+/// Flutter application.
+/// These are for example events related to its lifecycle, accessibility
+/// features, display features and more.
+///
+/// This class calls `WidgetsBinding.instance` but we don't need to call
+/// `WidgetsFlutterBinding.ensureInitialized()` because we can only add this
+/// class to an instance of `WidgetsBinding`.
+///
+/// See also:
+///   - [WidgetsBindingObserver](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html)
+class SentryWidgetsBindingObserver with WidgetsBindingObserver {
+  SentryWidgetsBindingObserver({Hub hub}) {
+    this.hub = hub ?? HubAdapter();
+  }
+
+  Hub hub;
+
+  /// This method records lifecycle events.
+  /// It tries to mimic the behavior of ActivityBreadcrumbsIntegration of Sentry
+  /// Android for lifecycle events.
+  ///
+  /// On Android and iOS this records lifecycle event breadcrumbs which loosely
+  /// correspond to their respectiv platforms lifecycle events.
+  ///
+  /// Does this need to be tracked? The nativ Android an iOS integration
+  /// also tracks this but what about Web, Linux, Windows and MacOS?
+  /// The message could include that this is a Flutter message.
+  ///
+  /// See also:
+  ///   - [WidgetsBindingObserver](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html)
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    // According to
+    // https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
+    // this is of the type navigation.
+    hub.addBreadcrumb(Breadcrumb(
+      message: _lifecycleToString(state),
+      category: 'ui.lifecycle',
+      type: 'navigation',
+    ));
+  }
+
+  /// Called when the application's dimensions change. For example,
+  /// when a phone is rotated or an application window is resized.
+  ///
+  /// See also:
+  ///   - [Window.onMetricsChanged](https://api.flutter.dev/flutter/dart-ui/Window/onMetricsChanged.html)
+  @override
+  void didChangeMetrics() {
+    // Should this method push a new scope with a new Device Context?
```

Comment:
not really, every time one does `captureException`, event or message, we read the fresh state of the device and set to the Contexts automatically, so ideally, this will create only a breadcrumb with the changes.

A bit similar to https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/AppComponentsBreadcrumbsIntegration.java#L75-L96
but there, we don't track the height and width size, but only that it has changed.

Does this API work for macOS/Linux and Web?

Reply:
Yes, this API works at least on macOS, Linux and [Windows](https://github.com/flutter/engine/blob/master/shell/platform/windows/win32_flutter_window.cc#L120). I don't know about web but I would guess so.

Reply:
thats cool, so I think it'd make sense to align the automatic breadcrumbs we already have on the Native SDKs and only add to this PR the ones that are not tracked by the native SDKs, so we remove the duplication.
The idea is to keep it simple, only adding "new" breadcrumbs.
Once we support all the other platforms, we could do the opposite, turning it off on the Native SDKs and enabling it here, what do you think?